extensions/websockets-next/server/runtime/src/main/java/io/quarkus/websockets/next/MessageCodec.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.websockets.next;
+
+import java.lang.reflect.Type;
+
+/**
+ * Used to encode and decode messages.
+ * <p>
+ * Implementation classes must be CDI beans. Qualifiers are ignored.
+ *
+ * @param <T>
+ * @param <MESSAGE>
+ */
+public interface MessageCodec<T, MESSAGE> {
+
+    /**
+     *
+     * @param type the type to handle, must not be {@code null}
+     * @return {@code true} if this codec can handle the provided type, {@code false} otherwise
+     */
+    boolean canHandle(Type type);
```

Comment:
I propose renaming this to `supports`

Reply:
That definitely sounds better. We may also consider `appliestTo()`...?

Reply:
Yeah, that's fine as well, but I find `support` provides the best way of explaining that this class will be _considered_ if the method returns `true`, and not necessarily _chosen_.

Reply:
Like it. Will change it later today ;-).

Reply:
🙏🏼